pkgs/build-support/mkshell/minimal.nix

```diff
@@ -0,0 +1,47 @@
+{ writeTextFile, writeScript, bash }:
+
+# A special kind of derivation that is only meant to be consumed by the
+# nix-shell. This differs from the traditional `mkShell` in that it is
+# ultra minimalist in dependencies.
+# It does not come with traditional stdenv (i.e. coreutils).
+# Its only dependency is essentially bash.
+{
+# a list of packages to add to the shell environment
+# we simplify here rather than `native` or `propagated` since
+# this is only ever used to be in a nix-shell; which is by default
+# the current system
+packages ? [ ], ... }@attrs:
+
+derivation ({
+  name = "minimal-nix-shell";
+
+  # What is the need for this reference to stdenv?
+  # Well Nix itself, checks to see if this attribute exists
+  # and sources the file.
+  # We will setup our nix-shell environment here and do things
+  # like clear the $PATH
+  # reference: https://github.com/NixOS/nix/blob/94ec9e47030c2a7280503d338f0dca7ad92811f5/src/nix-build/nix-build.cc#L494
+  "stdenv" = writeTextFile rec {
+    name = "setup";
+    executable = true;
+    destination = "/${name}";
+    text = ''
+      set -e
+      unset PATH
```

Comment:
Clearing PATH is not always desirable. Some commands are themselves a more stable interface than the interfaces your pinned command may misuse. Nix itself is one example of this. The nix daemon protocol isn't as stable as the CLI. Some commands should not be pinned at all. For example, a web application's shell may launch the user's browser as a convenience. This should not be pinned because of the overhead of duplicating browser packages and it won't have an effect because the browser will contact the user's running instance. If it doesn't, that's even worse because now the user may unwittingly keep using a version they didn't intend to use. Or maybe it has a forward compatibility bug, etc.

Reply:
What I would like to do is the following:
1. Prepend the _packages_ to the $PATH **if not pure**
2. Clear $PATH and only add _packages_ to $PATH **if pure**
> 1 & 2 match the current stdenv.mkShell 
(If in _pure_ mode, for instance, you don't have access to _nix_ CLI)

I am still looking to discover how _mkShell_ handles $PATH and do an imitation for when in pure mode.

Reply:
nix-shell clears the environment for you when `--pure`, so I don't think you need to unset it at all.

Reply:
Figured it out, please see linked Nix issue.